File: java/src/main/java/com/acqusta/tquant/api/TQuantApi.java
package com.acqusta.tquant.api;

import com.acqusta.tquant.api.impl.DataApiImpl;
import com.acqusta.tquant.api.impl.TradeApiImpl;
import com.acqusta.tquant.api.impl.TQuantApiImpl;

public class TQuantApi {

    public static void setParams(String key, String value)
    {
        TQuantApiImpl.setParams(key, value);
    }

    /**
     * 创建 TradeApi
     * @param addr 服务器地址，取值通常为 tcp://127.0.0.1:10082
     */
    public static TradeApi createTradeApi(String addr) throws Exception{
        return new TradeApiImpl(addr);
    }

    /**
     * 创建 DataApi
     *
     * @return
     */
    public static DataApi  createDataApi(String addr) throws Exception{
        return new DataApiImpl(addr);
    }
}

File: cs/src/api/TQuantApi.cs
using System;
using TQuant.Api.Impl;

namespace TQuant
{
    namespace Api
    {
        public class TQuantApi
        {
            public static void SetParams(string key, string value)
            {
                TqapiDll.tqapi_set_params(key, value);
            }

            public static TradeApi CreateTradeApi(string addr)
            {
                var h = TqapiDll.tapi_create(addr);
                return h != IntPtr.Zero ? new TradeApiImpl(h, true) : null;
            }

            public static DataApi CreateDataApi(string addr)
            {
                var h = TqapiDll.dapi_create(addr);
                return h != IntPtr.Zero ? new DataApiImpl(h, true) : null;
            }
        }
    }
}
